pynicotine/gtkgui/dialogs/preferences.py

```diff
@@ -671,37 +656,62 @@ def on_add_shared_folder(self, *_args):
     def on_edit_shared_folder_response(self, dialog, _response_id, iterator):
 
         new_virtual_name = dialog.get_entry_value()
-        new_is_buddy_only = dialog.get_option_value()
+        new_available_to = dialog.get_second_entry_value()
 
-        self._edit_shared_folder(iterator, new_virtual_name, new_is_buddy_only)
+        virtual_name = self.shares_list_view.get_row_value(iterator, "virtual_name")
+        available_to = self.shares_list_view.get_row_value(iterator, "available_to")
+
+        if new_virtual_name == virtual_name and new_available_to == available_to:
+            return
+
+        self.rescan_required = True
+
+        folder_path = self.shares_list_view.get_row_value(iterator, "folder")
+        group_names = {
+            _("Everyone"): "public",
+            _("Buddies"): "buddy",
+            _("Trusted buddies"): "trusted"
+        }
+        group_name = group_names.get(new_available_to)
+
+        core.shares.remove_share(
+            virtual_name, share_groups=(self.shared_folders, self.buddy_shared_folders, self.trusted_shared_folders)
+        )
+        new_virtual_name = core.shares.add_share(
+            folder_path, group_name=group_name, virtual_name=new_virtual_name,
+            share_groups=(self.shared_folders, self.buddy_shared_folders, self.trusted_shared_folders),
+            validate_path=False
+        )
+
+        self.shares_list_view.set_row_value(iterator, "virtual_name", new_virtual_name)
+        self.shares_list_view.set_row_value(iterator, "available_to", new_available_to)
 
     def on_edit_shared_folder(self, *_args):
 
         for iterator in self.shares_list_view.get_selected_rows():
             virtual_name = self.shares_list_view.get_row_value(iterator, "virtual_name")
             folder_path = self.shares_list_view.get_row_value(iterator, "folder")
-            is_buddy_only = self.shares_list_view.get_row_value(iterator, "buddy_only")
+            default_item = self.shares_list_view.get_row_value(iterator, "available_to")
 
             EntryDialog(
                 parent=self.application.preferences,
                 title=_("Edit Shared Folder"),
                 message=_("Enter new virtual name for '%(dir)s':") % {"dir": folder_path},
                 default=virtual_name,
-                option_value=is_buddy_only,
-                option_label=_("Share with buddies only"),
+                second_default=default_item,
+                second_droplist=[
+                    _("Everyone"),
```

Comment:
New unique term "Everyone" is inconsistent with the language used elsewhere, eg in menus "Public".